src/renderer/src/contexts/persistedState/createPersistedState.tsx

```diff
@@ -0,0 +1,62 @@
+import { createContext, useContext, useState, type ReactNode } from 'react'
+import { createStore, useStore, type StateCreator } from 'zustand'
+import { persist } from 'zustand/middleware'
+
+type PersistedStoreKey = 'ActiveProjectId'
+
+/**
+ * Follows the pattern of injecting persisted state with a context. See
+ * https://tkdodo.eu/blog/zustand-and-react-context. Allows for easier testing
+ */
+export function createPersistedStoreWithProvider<T>(
+	slice: StateCreator<T>,
+	persistedStoreKey: PersistedStoreKey,
+) {
+	const persistedStore = createPersistedStore(slice, persistedStoreKey)
+	// used for testing and injecting values into testing environment
+	const nonPersistedStore = createStore(slice)
+	// type persistedStore is a subset type of type nonPersistedStore
+	const Context = createContext<typeof nonPersistedStore | null>(null)
+
+	const Provider = ({ children }: { children: ReactNode }) => {
+		const [storeInstance] = useState(() => persistedStore)
+
+		return <Context.Provider value={storeInstance}>{children}</Context.Provider>
+	}
+
+	const useStoreHook = <Selected,>(
+		selector: (state: T) => Selected,
+	): Selected => {
+		const contextStore = useContext(Context)
+		if (!contextStore) {
+			throw new Error(
+				`Missing provider for persisted store: ${persistedStoreKey}`,
+			)
+		}
+
+		return useStore(contextStore, selector)
+	}
+
+	return { Provider, useStoreHook, Context, nonPersistedStore }
```

Comment:
I'm exposing the `Context` and the `Provider` here. The provider should be used by the app, the Context on the other hand can be used for testing to simulate the provider. This allows set up the same provider, without having to touch the persisted state, and injecting test values.

You can see an example of that [here](https://github.com/digidem/comapeo-desktop/blob/644cb2e8679f3ddb2abd9bbd4f42e94656c5804f/src/renderer/src/routes/index.test.tsx#L21). The Value of persisted state determines where the user is redirected to. So I can easily simulate that state with a non persisted value by using the context and nonPersistedStore

There could be an argument that we should use persisted state in the tests, to better mimic the app. The problem is that we would be overwriting the actual persisted state, which might not be ideal for developer experience. We could use a different "key". Therefore testing environments would have its own persisted state, separate from the dev environment. This could get a little buggy as there could be things in persisted state that the test environment is using that we don't realize. We could make sure to clear persisted state at the end of every test file, but there would be some overhead with that.

@gmaclennan, @achou11, and @cimigree I'm curious if you have any insight or opinions of the best approach for testing persisted state.